src/plugins/components/PluginSettings/PluginSettings.tsx

```diff
@@ -0,0 +1,84 @@
+import Card from "@material-ui/core/Card";
+import CardContent from "@material-ui/core/CardContent";
+import TextField from "@material-ui/core/TextField";
+import makeStyles from "@material-ui/styles/makeStyles";
+import CardTitle from "@saleor/components/CardTitle";
+import ControlledSwitch from "@saleor/components/ControlledSwitch";
+import { FormErrors } from "@saleor/types";
+import { ConfigurationTypeFieldEnum } from "@saleor/types/globalTypes";
+import React from "react";
+import i18n from "../../../i18n";
+import { FormData } from "../PluginsDetailsPage";
+
+interface PluginSettingsProps {
+  data: FormData;
+  errors: FormErrors<"name" | "configuration">;
+  disabled: boolean;
+  onChange: (event: React.ChangeEvent<any>) => void;
+  plugin: Array<{
```

Comment:
It should be named `fields`, `plugin` is very confusing since it's not plugin info.